modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/graphql/dto/v1_0/ObjectDefinitionGraphQLDTOContributor.java

```diff
@@ -154,11 +154,23 @@ else if (Objects.equals(
 							Map.class));
 				}
 
-				graphQLDTOProperties.add(
-					GraphQLDTOProperty.of(
-						objectField.getName(),
-						_typedClasses.getOrDefault(
-							objectField.getDBType(), Object.class)));
+				String objectFieldName = objectField.getName();
+
+				if (!Objects.equals(objectFieldName, "status")) {
+					graphQLDTOProperties.add(
+						GraphQLDTOProperty.of(
+							objectField.getName(),
+							_typedClasses.getOrDefault(
+								objectField.getDBType(), Object.class)));
+				}
+				else {
+					graphQLDTOProperties.add(
+						GraphQLDTOProperty.of(
+							objectField.getName(),
+							_typedClasses.getOrDefault(
+								ObjectFieldConstants.DB_TYPE_STRING,
+								Object.class)));
+				}
```

Comment:
Hey @JhossephG 
I believe it's easier to understand conditions when written in the "positive" form (avoiding negations) wherever possible. In this case, I would suggest restructuring the code like this:
```
if (Objects.equals(objectFieldName, "status")) {
		graphQLDTOProperties.add(
		GraphQLDTOProperty.of(
			objectField.getName(),
			_typedClasses.getOrDefault(
				ObjectFieldConstants.DB_TYPE_STRING,
				Object.class)));
}
else {
	graphQLDTOProperties.add(
		GraphQLDTOProperty.of(
			objectField.getName(),
			_typedClasses.getOrDefault(
				objectField.getDBType(), Object.class)));
}
```